src/Core/AbstractRequest.php

```diff
@@ -8,11 +8,20 @@
      */
     protected $client;
 
+    protected $asJson = false;
+
     /**
      * @param AbstractClient $client
      */
     public function __construct(AbstractClient $client)
     {
         $this->client = $client;
     }
+
+    public function asJson()
```

Comment:
I have now tried to do what you said, but there is a simple reason that it's not possible to do.. 

In somecases we cast to an array of objects. etc the battleTypes request.